src/serializer/Emitter.js

```diff
@@ -300,6 +301,11 @@ export class Emitter {
     }
 
     let val = ((dependencies: any): Value);
+    if (alreadyVisited.has(val)) {
+      return;
```

Comment:
`return undefined;` for consistency.
Leave behind a comment in the code why this is correct:
`// Note: If visiting this value earlier would have yielded a non-undefined value, then visiting would have been effectively stopped at that point, as the first non-undefined value is returned to the top-level. The fact that we got here again for an already visited value implies that we should return undefined.`